dvc/data/stage.py

```diff
@@ -247,6 +247,24 @@ def _get_staging(odb: "ObjectDB") -> "ObjectDB":
     return ReferenceObjectDB(fs, fs_path, state=state)
 
 
+def _load_untyped_dir_hash(odb, fs, fs_path, hash_info, meta, staging):
+    from .tree import Tree
+
+    obj = Tree.load(odb, hash_info.as_raw())
```

Comment:
This might raise `ObjectFormatError`, which is not how `_load_from_stage` behaved and so the code outside will not be ready for this